src/modules/cspec/geometry/CSPECGeometry.cpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2021 European Spallation Source, ERIC. See LICENSE file
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief CSPEC geometry class
+/// Based on CSPEC ICD documen thttps://project.esss.dk/owncloud/index.php/f/14482406
+
+/// Mapping from digital identifiers to x-, z- and y- coordinates
+//===----------------------------------------------------------------------===//
+
+#include <cspec/geometry/CSPECGeometry.h>
+#include <utility>
+#include <cmath>
+
+
+uint16_t Cspec::CSPECGeometry::xAndzCoord(uint8_t FENID, uint8_t HybridID, uint8_t VMMID, uint8_t Channel, uint16_t XOffset, bool Rotated){
+	if (!validWireMapping(HybridID, VMMID, Channel)){
+		XTRACE(DATA, WAR, "Invalid Hybrid: %u, VMM: %u, and Channel: %u, combination for wire mapping", HybridID, VMMID, Channel);
+    	// return std::pair<uint8_t, uint8_t>(Cspec::CSPECGeometry::InvalidCoord, Cspec::CSPECGeometry::InvalidCoord);
+    	return 65535;
+  	}
+	
+	//Wire equation defined in CSPEC ICD Document
+	uint16_t Wire = (HybridID * 2 + VMMID) * 64 + Channel - 32;
+
+	//odd FENs are second column in a vessel, 6 * 16 wires over
+	if (FENID % 2){
+		Wire = Wire + 6 * 16;
+	}
+
+	//when rotated local X coordinate is flipped around centre, ie 11-XCoord
+	//Z coord remains the same
+	//Then turning value back into wire value/combination of X and Z
+	if (Rotated){
+		uint8_t LocalXCoord = floor(Wire/16);
+		uint8_t LocalZCoord = Wire % 16;
+		LocalXCoord = 11 - LocalXCoord;
+		Wire = 16 * LocalXCoord + LocalZCoord;
+	}
+
+	Wire += 16 * XOffset;
+	return Wire;
+}
+
+uint8_t Cspec::CSPECGeometry::yCoord(uint8_t HybridID, uint8_t VMMID, uint8_t Channel, uint16_t YOffset, bool Rotated, bool Short){
+	uint8_t YCoord;
+	if (!validGridMapping(HybridID, VMMID, Channel, Short)){
+		XTRACE(DATA, ERR, "Invalid combination of HybridID: %u, VMMID: %u, Channel: %u, Short: %d", HybridID, VMMID, Channel, Short);
+		return 255;
```

Comment:
Change yCoord to uint16_t instead of uint8_t and share the same INVALID_COORDINATE const